server/routes.go

```diff
@@ -1529,6 +1530,24 @@ func (s *Server) ChatHandler(c *gin.Context) {
 		defer close(ch)
 		var sb strings.Builder
 		var toolCallIndex int = 0
+		var templateToolToken string
+		var tmpl *gotmpl.Template
+		if len(req.Tools) > 0 {
+			var ok bool
+			templateToolToken, ok = ToolToken(m.Template.Template)
+			if !ok {
+				slog.Debug("no tool token found")
+			}
+			tmpl, ok = ToolTemplate(m)
+			if !ok {
+				slog.Debug("no tool template found")
+			}
+		}
+
+		checkToolCall := false
+		if len(req.Tools) > 0 {
```

Comment:
This if statement is repeated above and here although we should see if we can eliminate both